osu.Game/Graphics/Containers/LinkFlowContainer.cs

```diff
@@ -37,7 +37,7 @@ public void AddLinks(string text, List<Link> links)
 
             foreach (var link in links)
             {
-                AddText(text.Substring(previousLinkEnd, link.Index - previousLinkEnd));
+                AddText(text[previousLinkEnd..link.Index]);
```

Comment:
am i alone in thinking this reads really badly?

Reply:
It's much safer this way, though. Very easy to get into off-by-one traps with index+length combos.

Reply:
ok, let's go with it for now. looks alright with font ligatures.